app/src/main/java/nu/yona/app/ui/LaunchActivity.java

```diff
@@ -115,7 +115,7 @@ else if (!getSharedUserPreferences().getBoolean(PreferenceConstant.STEP_PASSCODE
 			bundle.putInt(AppConstant.COLOR_CODE, ContextCompat.getColor(LaunchActivity.this, R.color.grape));
 			startNewActivity(bundle, PasscodeActivity.class);
 		}
-		else if (!TextUtils.isEmpty(getSharedUserPreferences().getString(PreferenceConstant.YONA_PASSCODE, "")))
+		else if (!TextUtils.isEmpty(getSharedUserPreferences().getString(PreferenceConstant.YONA_DATA, "")))
```

Comment:
Why is this?

Reply:
There should not be much difference. But, once I faced issue after un-subscribing. Later I was unable to reproduce. I suspect some thing with cache and I am more sure 'YONA_DATA' is cleared from cache.

Reply:
I suspect, this could be the reason for issue in the Vishal's phone, I am planning to create separate create a PR with extra logging around this and test in Vishal's phone.

Reply:
@Bert-R , can I add the needed logging in this PR?

Reply:
Yes, no problem